annotations.md

```diff
@@ -0,0 +1,228 @@
+            dbConfiguration: database-config  # configmap 
+            dbCredentials: database-cred-Secret # Secret
+            url: db.stage.ibm.com
+```
+
+
+
+### Scenarios
+
+
+1. #### Use everything from the Secret  “status.data.dbCredentials” 
+
+    Requirement : *Extract an entire configmap/Secret from the Kubernetes resource*
+
+
+    Annotation:
+
+    ```
+    “servicebinding.dev/dbcredentials”:”path={.status.data.dbcredentials},objectType=Secret”
+    ```
+
+
+    Descriptor:
+
+    ```
+    - path: data.dbcredentials
+      x-descriptors:
+        - urn:alm:descriptor:io.kubernetes:Secret 
+        - servicebinding
+    ```
+
+
+2. #### Use everything from the ConfigMap “status.data.dbConfiguration” 
+
+
+    Requirement : *Extract an entire configmap/Secret from the Kubernetes resource*
+
+    Annotation
+
+    ```
+    “servicebinding.dev/dbConfiguration”: "path={.status.data.dbConfiguration},objectType=ConfigMap”
+    ```
+
+
+    Descriptor
+
+    ```
+    - path: data.dbConfiguration
+      x-descriptors:
+        - urn:alm:descriptor:io.kubernetes:ConfigMap 
+        - servicebinding
+    ```
+
+3. #### Use “certificate” from the ConfigMap “status.data.dbConfiguration” as an environment variable
+
+    Requirement : *Extract a specific field from the configmap/Secret from the Kubernetes resource and use it as an environment variable.*
+
+
+    Annotation
+
+    ```
+    “servicebinding.dev/certificate”:
+    "path={.status.data.dbConfiguration},objectType=ConfigMap"
+    ```
+
+
+    Descriptor
+
+
+    ```
+    - path: data.dbConfiguration
+      x-descriptors:
+        - urn:alm:descriptor:io.kubernetes:ConfigMap 
+        - servicebinding:certificate:bindAs=volume
+    ```
+
+
+4. #### Use “certificate” from the ConfigMap “status.data.dbConfiguration” as a volume mount
+
+    Requirement : *Extract a specific field from the configmap/Secret from the Kubernetes resource and use it as a volume mount.*
+
+
+    Annotation
+
+    ```
+    “servicebinding.dev/certificate”:
+    "path={.status.data.dbConfiguration},bindAs=volume,objectType=ConfigMap"
+    ```
+
+
+    Descriptor
+
+
+    ```
+    - path: data.dbConfiguration
+      x-descriptors:
+        - urn:alm:descriptor:io.kubernetes:ConfigMap 
+        - servicebinding:certificate:bindAs=volume
+    ```
+
+
+5. #### Use “db_timeout” from the ConfigMap “status.data.dbConfiguration” as “timeout” in the binding Secret.
+
+    Requirement: *Extract a specific field from the configmap/Secret from the Kubernetes resource and map it to different name in the binding Secret*
+
+    Annotation
+
+    ```
+    “servicebinding.dev/timeout”: 
+    “path={.status.data.dbConfiguration},objectType=ConfigMap,sourceKey=db_timeout”
+    ```
+
+
+    Descriptor
+
+    ```
+    - path: data.dbConfiguration
+      x-descriptors:
+        - urn:alm:descriptor:io.kubernetes:ConfigMap 
+        - servicebinding:timeout:sourceKey=db_timeout
+    ```
+
+6. #### Use the attribute “status.data.url”
```

Comment:
we used both `url` and `uri` in this example.  I think you meant to use `uri` in all cases.

Reply:
yup!